source/contracts/trading/FillOrder.sol

```diff
@@ -389,7 +390,7 @@ contract FillOrder is CashAutoConverter, ReentrancyGuard, IFillOrder {
         }
 
         // AUDIT: is there a reentry risk here?  we execute all of the above code, which includes transferring tokens around, before we mark the order as filled
-        _tradeData.contracts.orders.fillOrderLog(_tradeData.order.orderId, _tradeData.filler.participantAddress, _tradeData.getMakerSharesDepleted(), _tradeData.getMakerTokensDepleted(), _tradeData.getFillerSharesDepleted(), _tradeData.getFillerTokensDepleted(), _tradeGroupId);
+        _tradeData.contracts.orders.fillOrderLog(_tradeData.order.orderId, _tradeData.filler.participantAddress, _tradeData.getMakerSharesDepleted(), _tradeData.getMakerTokensDepleted(), _tradeData.getFillerSharesDepleted(), _tradeData.getFillerTokensDepleted(), _settlementFees, _tradeGroupId);
```

Comment:
Do we not have enough data available to log directly here?  Going through two different contracts to log is quite unfortunate.  :/

Reply:
Yea we have to pull the order data from storage from the passed in ID. Without doing that IIR it's a stack depth issue